src/module_2/module_2_data.md

```diff
@@ -0,0 +1,1750 @@
+      <td>2021-04-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>085d8e598139ce6fc9f75d9de97960fa9e1457b409ec00...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-02 13:50:06</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+```python
+#Let's check the distribution of the number of regular products per user
+df_users_orders['num_regulars'].hist(bins=50, grid=False, color='#86bf91', log=True)
+
+plt.title('Number of regular products per user')
+plt.xlabel('num_regulars')
+plt.ylabel('Counts')
+plt.show()
+```
+
+
+    
+![png](module_2_data_files/module_2_data_22_0.png)
+    
+
+
+
+```python
+df_users_orders['num_regulars'].describe()
+```
+
+
+
+
+    count    4983.000000
+    mean        2.481437
+    std         8.890588
+    min         0.000000
+    25%         0.000000
+    50%         0.000000
+    75%         1.000000
+    max       320.000000
+    Name: num_regulars, dtype: float64
+
+
+
+
+```python
+#Percentage of users with more than 5 regular products
+len(df_users_orders[df_users_orders['num_regulars'] > 5])/len(df_users_orders)
+```
+
+
+
+
+    0.109773228978527
+
+
+
+
+```python
+# Mean number of regular products per user segment
+grouped = df_users_orders.groupby('user_segment')['num_regulars'].mean()
+
+grouped.plot(kind='bar')
+
+plt.title('Avg. Number of Regular Products per User')
+plt.xlabel('User Segment')
+plt.ylabel('Avg. Number of Regular Products')
+plt.show()
+```
+
+
+    
+![png](module_2_data_files/module_2_data_25_0.png)
+    
+
+
+
+```python
+# Family statistics
+num_not_null = df_users_orders['count_people'].notnull().sum()
+
+info  = num_not_null / len(df_users_orders) * 100
+
+num_children = len(df_users_orders[df_users_orders['count_children'] > 0]) / num_not_null * 100
+num_babies = len(df_users_orders[df_users_orders['count_babies'] > 0]) / num_not_null * 100
+num_pets = len(df_users_orders[df_users_orders['count_pets'] > 0]) / num_not_null * 100
+
+print(f"Percentage of family information: {info}%")
+
+print(f"Percentage of users with children: {num_children}%")
+print(f"Percentage of users with babies: {num_babies}%")
+print(f"Percentage of users with pets: {num_pets}%")
+```
+
+    Percentage of family information: 6.522175396347582%
+    Percentage of users with children: 40.0%
+    Percentage of users with babies: 7.076923076923077%
+    Percentage of users with pets: 40.61538461538461%
+
+
+### Conclusions
+
+<ol>
+<li>Not a lot of users have regular products (~ 90% have less than 5 products)</li>
+<li>There are outliers in 'num_regulars' (max. 320 regulars)</li>
+<li>Proposition users have more than double the number of regulars than Top Up users</li>
```

Comment:
Can you guess why is the case?

Reply:
I guess people don't have a lot of regulars because they don't buy that many products.
Relating to outliers, I can't think of a reason for having 320 regulars, maybe a wholesale buyer that has a lot of regulars set up.
For the last point, is it because the proposition users are the most frequent buyers and the top-up customers are just occasional buyers?